### PR TITLE
JAMES-2021 Switch to SLF4J for LocalDelivery related mailets

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SimpleMailStore.java [hunk 1]
```diff
@@ -100,7 +95,7 @@ public void storeMail(MailAddress recipient, Mail mail) throws MessagingExceptio
         mailboxAppender.append(mail.getMessage(), username, locatedFolder);
 
         metric.increment();
-        log.info("Local delivered mail " + mail.getName() + " successfully from " + DeliveryUtils.prettyPrint(mail.getSender())
+        LOGGER.info("Local delivered mail " + mail.getName() + " successfully from " + DeliveryUtils.prettyPrint(mail.getSender())
```
Comment (quynhn) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SimpleMailStore.java]: Pattern {} could be better?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SimpleMailStore.java [hunk 2]
```diff
@@ -115,7 +110,7 @@ private String computeUsername(MailAddress recipient) throws MessagingException
         try {
             return usersRepository.getUser(recipient);
         } catch (UsersRepositoryException e) {
-            log.warn("Unable to retrieve username for " + recipient.asPrettyString(), e);
+            LOGGER.warn("Unable to retrieve username for " + recipient.asPrettyString(), e);
```
Comment (quynhn) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SimpleMailStore.java]: Here also

**Comment by chibenwa** [comment]
@mbaechler it's not a design change, I did not touched to the mailet logging API, I just used SLF4J in a place where SLF4J was wrapped into another Commons-logging logger. It's not a design change IMO....

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**